tests/test_benchmark.py

```diff
@@ -41,6 +42,18 @@ class BenchmarkConfig:
     num_batches: int
     steps_per_batch: int
     verbose: bool
+    generate_plots: bool
+
+
+def plot_batch_times(batch_times: List[float], label: str):
+    plt.title(label)
+    plt.plot(batch_times)
+    plt.axhline(np.mean(batch_times), linestyle="--", c="gray", label="Mean")
+    plt.legend()
+    plt.xlabel("Batch")
+    plt.ylabel("Batch times (s)")
```

Comment:
images attached to pr description indicate we might want to add
```suggestion
    plt.ylabel("Batch times (s)")
    plt.tight_layout()
```